File: frontend/rick-and-morty-app/src/app/core/services/models/episode-filter.ts
export class EpisodeFilter {

    constructor(
        public name: string = '',
        public episode: string = ''
    ){}

    toQueryString(): string {
        const params: string [] = [];

        if(!!this.name.trim()) params.push(`name=${encodeURIComponent(this.name)}`)
        if(!!this.episode.trim()) params.push(`episode=${encodeURIComponent(this.episode)}`)

        return params.join('&');
    }

    hasFilters() : boolean {
        return !!this.name || !!this.episode;
    }
}

File: backend/PruebaTecnicaChileautos.Core/Filters/EpisodeFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PruebaTecnicaChileautos.Core.Filters
{
    public class EpisodeFilter
    {
        public string? Name { get; set; }
        public string? Episode { get; set; }


        public string ToQueryString()
        {
            var queryParams = new List<string>();

            if (!string.IsNullOrWhiteSpace(Name))
                queryParams.Add($"name={Uri.EscapeDataString(Name)}");

            if (!string.IsNullOrWhiteSpace(Episode))
                queryParams.Add($"episode={Uri.EscapeDataString(Episode)}");

            return string.Join("&", queryParams);
        }
    }
}
